controller/k8s/test_helper.go

```diff
@@ -33,10 +34,30 @@ func NewFakeAPI(configs ...string) (*API, error) {
 	), nil
 }
 
-type byPod []*corev1.Pod
+type hasUID interface {
+	GetUID() types.UID
+}
+
+type byUID []hasUID
+
+func (s byUID) Len() int { return len(s) }
+
+func (s byUID) Swap(i, j int) { s[i], s[j] = s[j], s[i] }
 
-func (s byPod) Len() int { return len(s) }
+func (s byUID) Less(i, j int) bool { return s[i].GetUID() < s[j].GetUID() }
 
-func (s byPod) Swap(i, j int) { s[i], s[j] = s[j], s[i] }
+func podByUID(pods []*corev1.Pod) byUID {
```

Comment:
This is necessary because it is not possible to treat a `[]*corev1.Pod` as a `[]hasUID` even though `Pod` implements `hasUID`.  I can only assume that this design decision arose from a deep hatred for humankind.

Reply:
that's really strange for me, I found some explanation [here](https://stackoverflow.com/questions/12990338/cannot-convert-string-to-interface/12990540#12990540) and [here](https://stackoverflow.com/a/12994852/5164435) to give some peace for my mind. Do you think, is it worth to comment something here explaining this decision?

Reply:
what about this pattern...
https://github.com/linkerd/linkerd2/blob/bdf5b46d093a3ff45f52b9538fee046d2449890e/controller/api/public/grpc_server_test.go#L33-L45